cogs/ReactForRole.py

```diff
@@ -0,0 +1,716 @@
+#!/usr/bin/env python
+
+"""
+KoalaBot Reaction Roles Code
+
+Commented using reStructuredText (reST)
+"""
+# Futures
+
+# Built-in/Generic Imports
+import re
+from typing import *
+
+import discord
+import emoji
+from discord.ext import commands
+
+# Own modules
+import KoalaBot
+from utils import KoalaDBManager, KoalaColours
+
+# Libs
+
+# Constants
+UNICODE_DISCORD_EMOJI_REGEXP: re.Pattern = re.compile("^:(\w+):$")
+CUSTOM_EMOJI_REGEXP: re.Pattern = re.compile("^<a?:(\w+):(\d+)>$")
+UNICODE_EMOJI_REGEXP: re.Pattern = re.compile(emoji.get_emoji_regexp())
+
+
+class ReactForRole(commands.Cog):
+    def __init__(self, bot: discord.Client):
+        self.bot = bot
+        KoalaBot.database_manager.create_base_tables()
+        KoalaBot.database_manager.insert_extension("ReactForRole", 0, True, True)
+        self.rfr_database_manager = ReactForRoleDBManager(KoalaBot.database_manager)
+        self.rfr_database_manager.create_tables()
+
+    @commands.group(name="reactForRole", aliases=["rfr", "react_for_role", "ReactForRole"])
+    async def react_for_role_group(self, ctx: commands.Context):
+        return
+
+    @commands.check(KoalaBot.is_admin)
+    @react_for_role_group.command(name="createMessage")
+    async def rfr_create_message(self, ctx: commands.Context):
+        await ctx.send(
+            "Okay, this will create a new react for role message in a channel of your choice."
+            "\nNote: The channel you specify will have its permissions edited to make it such that members are unable "
+            "to add new reactions to messages, they can only reaction with existing ones. Please keep this in mind, or"
+            " setup another channel entirely for this.")
+        channel_raw = await self.prompt_for_input(ctx, "channel ID, name or mention")
+        channel: discord.TextChannel = await commands.TextChannelConverter().convert(ctx, channel_raw)
```

Comment:
Recieved this error when waiting for timeout:
```
TypeError
expected string or bytes-like object```
Make sure if a timeout happens, there are no further errors passed to the user, a return when none may be ideal

Reply:
On travis?

Reply:
No, through manual testing, if you look at the bottom of #jack-test-bot you'll see the interaction